simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1002 @@
+        """
+        if self.telescope_indices is None:
+            self.num_of_hist = 1
+        else:
+            self.num_of_hist = len(self.telescope_indices)
+
+        self.hist_position, self.hist_direction, self.hist_time_altitude = [], [], []
+
+        for step in range(self.num_of_hist):
+            boost_axes_position = self._create_regular_axes("hist_position")
+            self.hist_position.append(
+                bh.Histogram(
+                    boost_axes_position[0],
+                    boost_axes_position[1],
+                    boost_axes_position[2],
+                )
+            )
+            boost_axes_direction = self._create_regular_axes("hist_direction")
+            self.hist_direction.append(
+                bh.Histogram(
+                    boost_axes_direction[0],
+                    boost_axes_direction[1],
+                )
+            )
+
+            boost_axes_time_altitude = self._create_regular_axes("hist_time_altitude")
+            self.hist_time_altitude.append(
+                bh.Histogram(
+                    boost_axes_time_altitude[0],
+                    boost_axes_time_altitude[1],
+                )
+            )
+
+    def _fill_histograms(self, photons, azimuth_angle=None, zenith_angle=None):
+        """Fill the histograms created by self._create_histogram
+
+        Parameters
+        ----------
+        photons: list
+            List of size M of numpy.void of size (N,8), where M is the number of telescopes in the
+            array, N is the number of photons that reached each telescope. The following information
+             of the Cherenkov photons on the ground are saved: x, y, cx, cy, time, zem, photons,
+             wavelength.
+        azimuth_angle: astropy.Quantity
+            Azimuth angle to rotate the observational plane and obtain it perpendicular to the
+            incoming event. It can be passed in radians or degrees.
+            If not given, no rotation is performed.
+        zenith_angle: astropy.Quantity
+            Zenith angle to rotate the observational plane and obtain it perpendicular to the
+            incoming event. It can be passed in radians or degrees.
+            If not given, no rotation is performed.
+
+        Raises
+        ------
+        IndexError:
+            If the index or indices passed though telescope_index are out of range.
+        """
+
+        hist_num = 0
+        for one_tel_info, photons_info in zip(self._tel_positions, photons):
+
+            if azimuth_angle is None or zenith_angle is None:
+                photon_x, photon_y = photons_info["x"], photons_info["y"]
+            else:
+                photon_x, photon_y = rotate(
+                    photons_info["x"],
+                    photons_info["y"],
+                    azimuth_angle,
+                    zenith_angle,
+                )
+
+            if self.telescope_indices is None:
+                hist_num = 0
```

Comment:
It is actually important to redefine it to zero. Because while I am iterating in the telescopes I am filling only the first histogram (there is only one histogram for each parameter whenever `self.telescope_indices is None`).

Reply:
So I suggest to do the opposite and increment `hist_num` only if `self.telescope_indices is not None`. It would make the code clearer and save you a command. Not sure 100% if it's possible, but I think so.